lib/linter.js

```diff
@@ -890,21 +898,25 @@ class Linter extends EventEmitter {
 
             const ecmaFeatures = this.currentConfig.parserOptions.ecmaFeatures || {};
             const ecmaVersion = this.currentConfig.parserOptions.ecmaVersion || 5;
+            const sourceType = this.currentConfig.parserOptions.sourceType || "script";
```

Comment:
I think it's a breaking change.

This line is copied from https://github.com/eslint/eslint/pull/8755/files/fbd75522568fa06aacfa8375bc971b79c83ae304#diff-98bf811b7b68f5089bfb2605df5b2548L900